Composer/packages/server/src/utility/attachLSP.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as http from 'http';
+import * as url from 'url';
+import * as net from 'net';
+
+import * as ws from 'ws';
+import * as rpc from 'vscode-ws-jsonrpc';
+
+export function createSocketHandler(webSocket: any): rpc.IWebSocket {
+  const socket: rpc.IWebSocket = {
+    send: content =>
+      webSocket.send(content, error => {
+        if (error) {
+          throw error;
```

Comment:
1. This exception will be caught by `webSocket.send`. Is it the intended audience?
2. According to MDN, `ws.send` only accept single parameter. The second parameter (error callback in your case), are never used.
3. `WebSocket.send` is for *queueing message to send*, it did not guarantee when the message will be sent. And there is no check for that.